### ISSUE TITLE
Train YoloV4 with negative examples

### ISSUE THREAD

**Issue by BenJam29** [AlexeyAB/darknet#6783]
Hi guys, 
I wanted  to train YoloV4 with negative examples as well. My question is that what the ratio would be for the created dataset? 
Suppose I have 2000 training images. If 500 images (25%) are background images without any desired objects, is it a proper ratio for training set or I should reduce or increase the size of negative examples?

Thank you,

**Comment by bulatnv88** [comment]
From my experience, negative samples don't improve accuracy.
Additionally, you should be 100% sure that there is no objects. Otherwise accuracy will drop down.

**Comment by stephanecharette** [comment]
> negative samples don't improve accuracy

I disagree.  Negative images *definitely* help!  And in industrial settings where you might have very low number of classes and very repetitive images, they are required otherwise the network wont train correctly.  If you have 2K images and 500 negative images, you are good.

**Comment by bulatnv88** [comment]
@stephanecharette Sorry, my falt. Yes, negeative samples should be very useful for small datasets. I have train 7 classes with 50K and 100K images.

**Comment by BenJam29** [comment]
Thank you very much @stephanecharette and @bulatnv88. I know the percentage is very experimental. I just wanted to benefit from other practitioners' experience.

**Comment by Rieder79** [comment]
A question from my side belonging negative samples: To use negative examples I use the according image with an empty txt-file for training, right?
Everything else just like in positive examples.

**Comment by stephanecharette** [comment]
> To use negative examples I use the according image with an empty txt-file for training, right?

Correct.

**Comment by personwhofloat** [comment]
Hello,
What if my dataset contains a static background, meaning it does not change a lot. How many negative samples should I include in my dataset?

**Comment by BenJam29** [comment]
> Hello,
What if my dataset contains a static background, meaning it does not change a lot. How many negative samples should I include in my dataset?

Hi, 
Based my experience, between 20-25% negative samples could be beneficial for detection performance. If the negative images are very similar visually, you might over fit to those images. Data augmentation could be a solution for to diversify your negative samples.  
Best,
